core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchange.java

```diff
@@ -141,7 +143,12 @@ else if (isScaledWriterHashDistribution(partitioning)) {
                     bufferCount,
                     1,
                     writerScalingMinDataProcessed.toBytes(),
-                    getSkewedPartitionMinDataProcessedRebalanceThreshold(session).toBytes());
+                    getSkewedPartitionMinDataProcessedRebalanceThreshold(session).toBytes(),
+                    // Keep the maxPartitionsToRebalance to atleast writer count such that single partition writes do
+                    // not suffer from skewness and can scale uniformly across all writers. Additionally, note that
+                    // maxWriterCount is calculated considering memory into account. So, it is safe to set the
+                    // maxPartitionsToRebalance to maximum number of writers.
```

Comment:
> So, it is safe to set the maxPartitionsToRebalance to maximum number of writers.

That kind of depends on order whether you will scale first or see all partitions first.